libs/category/src/models/filters/type/range/pair.ts

```diff
@@ -0,0 +1,8 @@
+import { DaffCategoryFilterOptionBase } from '../../category-filter-option-base';
+import { DaffCategoryFilterRangeOption } from './option';
+
+export interface DaffCategoryFilterRangePair<T> extends DaffCategoryFilterOptionBase {
+	applied: true;
```

Comment:
`applied: true` can be removed because of the `DaffCategoryFilterOptionBase` extension.

Reply:
`applied: true` is a distinct type constraint from `applied: boolean` though

Reply:
hmm, true. Seems like the `DaffCategoryFilterOptionBase` should be removed then, since there is no overlap between `DaffCategoryFilterRangePair` and `DaffCategoryFilterEqualOption`.

Reply:
It would be necessary if we needed to add future types of filters.

Reply:
It doesn't make sense for `DaffCategoryFilterRangePair` to extend a class it has nothing in common with. That extension should be removed, or the `DaffCategoryFilterOptionBase` should be changed.

I guess you could have the `DaffCategoryFilterOptionBase` have no fields so both the Equal and Range option model extend the same class and they can be referenced together throughout the code, but I figure that's what the `DaffCategoryFilterOption` is for.

Reply:
I disagree with this, there's subtyping here that's important. It's important to note that the typescript compiler isn't complaining as well.

```ts
type A = {
 prop: boolean;
}

interface B extends A {
 prop: boolean;
}

interface C extends A{
 prop: true
}
``` 

Type `A` describes the segregated interface for some operations that we will be performing on both `B` and `C`. 

Consider instead.

```ts
type A = B | C;

type B = {
 prop: boolean;
}

type C = {
 prop: true
}

const concreteA: A = {
  prop: false
}
```

This benefit here is in maintenance as we don't have to continuously append new types to `A` as we create them. We're being "open for extension, and closed for modification".

Reply:
It seems like you're arguing a strawman. I'm not saying we shouldn't have subtyping, or that subtyping is unimportant. I'm saying that this is false: "Type A describes the segregated interface for some operations **_that we will be performing on both B and C_**."

Your example assumes that C needs `prop: true` to operate. In our example, it doesn't. Consider this instead:
```
interface DaffCategoryFilterOptionBase {
   applied: boolean;
   value: string;
   label: string;
   count: number;
}

interface DaffCategoryFilterEqualOption extends DaffCategoryFilterOptionBase {}

interface DaffCategoryFilterRangePair extends DaffCategoryFilterOptionBase {
   applied: true;
   value: null;
   label: null;
   count: null;
   min: DaffCategoryFilterRangeOption<T>;
   max: DaffCategoryFilterRangeOption<T>;
}
```
Obviously, this makes no sense to do. But I fail to see why what you're suggesting is essentially any different.

Reply:
After further discussion, it was decided it is better to keep the `applied` field on the `DaffCategoryFilterRangePair` to make easier to work with other filter types that do need the `applied` field, e.g. in the `daffCategoryFindAppliedFilterOptions` function.